features/events/api/src/main/java/org/opennms/netmgt/xml/event/IEvent.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2020 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2020 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+package org.opennms.netmgt.xml.event;
+
+import java.net.InetAddress;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.List;
+
+/**
+ * A definition corresponding to POJO '{@link Event}'.
+ */
+public interface IEvent {
+    IAlarmData getAlarmData();
+    IAutoAcknowledge getAutoacknowledge();
+    IAutoAction getAutoaction(final int index) throws IndexOutOfBoundsException;
```

Comment:
Just wondering, is there any use in explicitly declaring the throws clause for a runtime exception? I see that the original bean did so, but does it actually provide any value?

Reply:
Yeah, it doesn't provide any value - so I removed it...